pkg/node/manager/manager.go

```diff
@@ -344,7 +348,7 @@ func (m *Manager) NodeUpdated(n nodeTypes.Node) {
 		// through the tunnel to preserve the source identity as part of the
 		// encapsulation.
 		if address.Type == addressing.NodeCiliumInternalIP || m.conf.NodeEncryptionEnabled() ||
-			option.Config.EnableHostFirewall {
+			option.Config.EnableHostFirewall || option.Config.JoinK8sNamespace != "" {
```

Comment:
Just judging from the option name, it might not be obvious that `option.Config.JoinK8sNamespace != ""` means "VM mode". Maybe add a helper function to `option.DaemonConfig` for this check which is more aptly named? This would also be useful in other places where we check the same condition.

Reply:
Maybe we need a more aptly named boolean option and then use `--k8s-namespace` in addition? Any suggestions for names? "VM mode" is confusing, IMO, as k8s nodes are also usually VMs, and the same mode can be used to join bare metal machines to Cilium clusters...

Reply:
I just saw that @aanm also mentioned this in https://github.com/cilium/cilium/pull/13309#discussion_r496199006. Agree that a new boolean "VM mode" option and using `--k8s-namespace` would be better in terms of user experience and code readability. And I also agree that having the option called "VM mode" might be confusing given VM is quite an overloaded term. Maybe @tgraf has a suggestion?

Reply:
Defining a new option `--join-cluster` and re-using existing `--k8s-namespace` instead.